### ISSUE TITLE
perhaps implementing the pairing logic inside this extension is not necessary

### ISSUE THREAD

**Issue by EnixCoda** [formulahendry/vscode-auto-rename-tag#61]
I've seen multiple issue complaining about bugs of pairing wrong tags under this repo and, for myself, have been running into those bugs a lot recently.

Maybe working out a better algorithm to find the correct pair costs too much effort so the bugs still remain. I'd like to thank you for providing this extension anyway!

I have one opinion about making this extension better, without working out a *perfect* algorithm by ourselves.

---

After disabling all my VS Code extensions, I found it can still find tag pairs for us(see highlighted).

![image](https://user-images.githubusercontent.com/7480839/41508392-992024b8-7276-11e8-9785-e0cd4af0998e.png)

![image](https://user-images.githubusercontent.com/7480839/41508390-968ca348-7276-11e8-921e-9a118c0bf98c.png)

So I think there should be API providing currently focused tag pairs! Find it out and use it to enhance this extension would be much easier.

**Comment by alystair** [comment]
I agree as the system this extension uses breaks more often than I'd like, sometimes replacing a parent start tag..
